source/solvers/vof.cc

```diff
@@ -463,7 +449,63 @@ VOF<dim>::postprocess(bool first_iteration)
 
 template <int dim>
 void
-VOF<dim>::pre_mesh_adaptation()
+VolumeOfFluid<dim>::modify_solution()
+{
+  if (this->simulation_parameters.multiphysics.interface_sharpening)
+    {
+      // Limit the phase fractions between 0 and 1
+      update_solution_and_constraints(present_solution);
+      for (unsigned int p = 0; p < previous_solutions.size(); ++p)
+        update_solution_and_constraints(previous_solutions[p]);
+
+      // Interface sharpening is done at a constant frequency
+      if (this->simulation_control->get_step_number() %
+            this->simulation_parameters.interface_sharpening
+              .sharpening_frequency ==
+          0)
+        {
+          if (simulation_parameters.linear_solver.verbosity ==
+              Parameters::Verbosity::verbose)
+            {
+              this->pcout << "Sharpening interface at step "
+                          << this->simulation_control->get_step_number()
+                          << std::endl;
+            }
+
+          const DoFHandler<dim> *dof_handler_fluid =
+            multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
```

Comment:
```suggestion
          const DoFHandler<dim> *dof_handler_fluid =
            multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
```
Do you really need to do this here? Why not do it in the assembly routine?